Assets/MixedRealityToolkit/Inspectors/Profiles/DataProviderAccessServiceInspector.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.﻿
+
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using System;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Editor
+{
+    /// <summary>
+    /// Abstract class providing base functionality for data provider management in inspector. Useful for core systems that follow dataprovider access model.
+    /// </summary>
+    public abstract class BaseDataProviderServiceInspector : BaseMixedRealityToolkitConfigurationProfileInspector
+    {
+        protected class ServiceConfigurationProperties
+        {
+            internal SerializedProperty componentName;
+            internal SerializedProperty componentType;
+            internal SerializedProperty providerProfile;
+            internal SerializedProperty runtimePlatform;
+        }
+
+        protected abstract SerializedProperty GetDataProviderConfigurationList();
```

Comment:
Nit: it might be good to document these abstract methods to help explain their implementation in new classes of this type

Reply:
Yea I'll make a pass through for this PR